README.md

```diff
@@ -201,6 +201,28 @@ This is an example of applying an auto-release policy for the product service fo
 $ hamctl policy --service example apply auto-release --branch master --env dev
 ```
 
+#### Branch restriction on environments
+
+A `branch-restriction` policy instructs the release manager to only allow artifacts from specific branches to be released to an environment.
+The `--matcher` flag defines a regular expression that is matched against the branch name on every release.
+
+As an example, the following command applies a branch-restriction policy for the `example` service that only allows the `master` branch to be released to the `prod` environment.
+
+```
+$ hamctl policy --service example apply branch-restriction --env prod --matcher '^master$'
```

Comment:
That's a good idea. It might be confusing in behaviour to the `auto-release` policy though.

What is the expected output of these four commands?

```
$ hamctl policy apply branch-restriction --env prod --branch 'master'
$ hamctl policy apply branch-restriction --env prod --branch 'hotfix'

$ hamctl policy apply auto-release --env prod --branch 'master'
$ hamctl policy apply auto-release --env prod --branch 'hotfix'
```

The branch-restriction will result in ONLY allowing branch `hotfix` in prod. The auto-releases will release any artifacts from `master` AND `hotfix` into prod.

I do however like the explicitness of the flag name so if we just change `matcher` to `branch-regex`, wouldn't that suffice?

Another solution could be to add the `branch` flag and require and a `--force` flag to overwrite? But then again, the behaviour is different.

Reply:
I kind of link `matcher` or `match-regex`. The policy name already tells what it does. 
`$ hamctl policy apply branch-restriction --env prod --match-regex 'master'` is, at least to me, very easy to understand. That's just my 2 cents 🤷‍♂

Reply:
Yeah, I can see it work both ways :) 
```
$ hamctl policy apply branch-restriction --env prod --branch-regex 'master'
$ hamctl policy apply branch-restriction --env prod --match-regex 'master'
```

Reply:
From a user's perspective it might be more familiar with `branch-regex`?

Further I just thought of a follow up. Currently `hamctl` will auto complete the branch name, but that would give a "non-strict" match. It would be nice if the autocomplete actually set the start and end identifiers.

```
$ hamctl policy apply branch-restriction --env prod --branch-regex [tab][tab]
master hotfix/fix
[enter]
$ hamctl policy apply branch-restriction --env prod --branch-regex '^master$'
```

Reply:
👍

Reply:
I've renamed to `branch-regex` and also updated all internal fields to follow this name. 👍

Reply:
Great! - with the auto-completion of the regex, it is also less likely the user will forget `^$`.